src/components/ResourceBrowser/ResourceList/K8SResourceList.tsx

```diff
@@ -189,15 +230,17 @@ export const K8SResourceList = ({
             setFilteredResourceList(null)
             return
         }
-        handleFilterChanges(searchText)
-    }, [resourceList])
+
+        handleFilterChanges(searchText, sortBy, sortOrder)
+        setResourceListOffset(0)
+    }, [resourceList, sortBy, sortOrder])
```

Comment:
pls check if same should be done for search as well

Reply:
It is already handled with onChange of search field